pengine/graph.c

```diff
@@ -1079,6 +1098,10 @@ action2xml(action_t * action, gboolean as_input, pe_working_set_t *data_set)
         add_downed_nodes(action_xml, action, data_set);
     }
 
+    if (safe_str_eq(action->task, CRM_OP_MAINTENANCE_NODES)) {
+        add_maintenance_nodes(action_xml, action, data_set);
+    }
+
```

Comment:
Doubting that this would really be beneficial from a performance pov.
Looping through a GList is really cheap compared to the xml-crafting done within the loop.
Guess from cache-perspective doing it in a closer loop might even be faster.
And from an architectural view we would move transition-xml-crafting code out of graph.c.
It is already not too beautiful to have code that creates an action in stage8 but it doesn't
really fit context-wise into any of the other stages either so I finally put it into stage8.